src/btree/bt_prefetch.c

```diff
@@ -25,11 +25,7 @@ __wt_btree_prefetch(WT_SESSION_IMPL *session, WT_REF *ref)
     conn = S2C(session);
     block_preload = 0;
 
-    /*
-     * Do not add refs from trees that have eviction disabled since they are probably being closed.
-     */
-    if (!(F_ISSET(ref, WT_REF_FLAG_LEAF)) || S2BT(session)->evict_disabled > 0 ||
-      (__wt_session_gen(session, WT_GEN_SPLIT) == 0)) {
+    if (!(F_ISSET(ref, WT_REF_FLAG_LEAF) || __wt_session_gen(session, WT_GEN_SPLIT) == 0)) {
```

Comment:
I'm not sure why we check the session gen here, if anyone knows please clarify

Reply:
Right we're traversing the internal page hence we need a split gen.

Reply:
Pre-fetch requires a split generation to traverse internal pages so we need `__wt_session_gen(session, WT_GEN_SPLIT) != 0`.

Reply:
@clarissecheah can you update the comment in the code now that we've moved the eviction check, explaining why we need a leaf page and a split gen.